allennlp/models/model.py

```diff
@@ -85,6 +86,32 @@ def forward(self, input1, input2, targets=None):
         """
         raise NotImplementedError
 
+    def forward_on_instance(self, instance: Instance) -> Dict[str, numpy.ndarray]:
+        """
+        Takes an :class:`~allennlp.data.instance.Instance`, which typically has raw text in it,
+        converts that text into arrays using this model's :class:`Vocabulary`, passes those arrays
+        through :func:`self.forward()`, and returns the result.  Before returning the result, we
+        convert any ``torch.autograd.Variables`` or ``torch.Tensors`` into numpy arrays and remove
+        the batch dimension.
+        """
+        # Hack to see what cuda device the model is on, so we know where to put these inputs.  For
+        # complicated models, or machines with multiple GPUs, this will not work.  I couldn't find
+        # a way to actually query what device a tensor / parameter is on.
+        cuda_device = 0 if next(self.parameters()).is_cuda else -1
```

Comment:
is it possible to have a pytorch module with no parameters? because wouldn't this crash in that case? or is that not worth worrying about

Reply:
Yes, it's possible.  But this is on `Model`, and if you don't have any parameters in your model, why are you using this library?